### DISCUSSION TITLE
Postfix issues

### DISCUSSION THREAD

**Discussion by username0136** [dani-garcia/vaultwarden#2560] [Q&A] [answered]
I've my postfix setup on my server but for some reason vaultwarden gets connection refused and nextcloud is able to send the mail so can anyone please help me figure out what's wrong?
Nextcloud configuration that works fine
![image](https://user-images.githubusercontent.com/102938585/173734774-97121f2a-f9f2-4d0b-8de6-be2645a7f855.png)

VaultWarden configuration that throws connection refused error even when mynetworks is allowing localhost
![image](https://user-images.githubusercontent.com/102938585/173734879-206d09f7-391c-471b-b383-9168a2f2acbc.png)

**Comment by username0136** [comment, marked as answer]
ok so for some reason docker can't access localhost so as a workaround I used my system local ip like 192.168.1.3 and it worked fine :-)

**Reply by BlackDex** [reply]
localhost for a docker container is within the container it self (unless you change the networking config of the container).
So by default a container always needs an external IP, or when using the Docker networking you could use the container name of the other containers, which will result in the external IP of the container.

**Reply by username0136** [reply]
:pray: Thanks for the kind explanation
